File: pset1/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>


float UserInput(void);
int Convert(float Dollar);
void Change(int Cents);


int main(void)
{
    float Dollar = UserInput();
    int Cents = Convert(Dollar);
    Change(Cents);
}


// Checks User Input
float UserInput(void)
{
    float Input;
    do
    {
        Input = get_float("Change owed: "); 
    }
    while (Input < 0 || Input == 0);
    return Input;
}


// Convert Dollars to Cents
int Convert(float Dollar)
{
    int Cent = round(Dollar * 100);
    return Cent;
}

// Counts Change
void Change(int Cents)
{
    int Quarter,Dime,Nickel,Penny,Sum;
    Quarter = Cents / 25;
    Dime = Cents % 25 / 10;
    Nickel = Cents % 25 % 10 / 5;
    Penny = Cents % 25 % 10 % 5;
    Sum = Quarter + Dime + Nickel + Penny;
    printf("%d\n", Sum);
}

File: pset6/cash (python)/cash.py
from cs50 import get_float

# user input
while True:
    change = get_float("Change owed: ")
    if change >= 0:
        break

#calculations
tup1 = (25, 10, 5, 1)
n = 0
coins = 0
change = change * 100
while True:
    while change >= tup1[n]:
        change = change - tup1[n]
        coins += 1
    n += 1
    if n == 4:
        break
print(f"{coins}")